File: XMLCheck/Handler/CheckRulesXML01/Check_GT_THE_TU.cs
using XMLCheck.Handler.CheckRulesXML1;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML01
{
    public class Check_GT_THE_TU : Xml1RuleBase
    {
        public override string Key => "GT_THE_TU";

        public override ErrorDetails? Check(Xml1Model model, HoSoContext context)
        {
            // 1. Không được để trống
            if (model.GT_THE_TU == null)
            {
                return Error("GT_THE_TU không được để trống");
            }
            if (string.IsNullOrWhiteSpace(model.MA_THE_BHYT))
            {
                return null;
            }

            var bhytList = model.MA_THE_BHYT
                .Split(';', StringSplitOptions.RemoveEmptyEntries)
                .Select(x => x.Trim())
                .ToList();
            if (bhytList.Count > 1)
            {
                return Error("Thứ tự GT_THE_TU không khớp với thứ tự của MA_THE_BHYT");
            }

            return null;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml1/Check_GT_THE_TU.ts
import { Xml1RuleBase } from './xml1RuleBase';
import { Xml1Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';

export class Check_GT_THE_TU extends Xml1RuleBase {
  constructor() {
    super({
      ruleId: 'RULE_XML1_GT_THE_TU_01',
      xmlType: 'XML1',
      field: 'GT_THE_TU',
      severity: 'ERROR',
      description: 'Kiểm tra quy định cho trường GT_THE_TU',
      errorMessage: 'GT_THE_TU không được để trống',
      reference: 'QĐ 3176/QĐ-BYT',
    });
  }

  

  public check(model: Xml1Model, context: HoSoContext): ErrorDetails | null {
    if (!model.GT_THE_TU) {
      return this.error('GT_THE_TU không được để trống');
    }

    if (!model.MA_THE_BHYT || !model.MA_THE_BHYT.trim()) {
      return null;
    }

    const bhytList = model.MA_THE_BHYT
      .split(';')
      .map(x => x.trim())
      .filter(x => x.length > 0);

    if (bhytList.length > 1) {
      return this.error("Thứ tự GT_THE_TU không khớp với thứ tự của MA_THE_BHYT");
    }

    return null;
  }
}
